client/finality-grandpa/src/lib.rs

```diff
@@ -1089,32 +1095,44 @@ pub fn setup_disabled_grandpa<Block: BlockT, Client, N>(
 /// Returns the key pair of the node that is being used in the current voter set or `None`.
 fn is_voter(
 	voters: &Arc<VoterSet<AuthorityId>>,
-	keystore: &Option<KeyStorePtr>,
-) -> Option<AuthorityPair> {
+	keystore: &Option<BareCryptoStorePtr>,
+) -> Option<AuthorityId> {
 	match keystore {
 		Some(keystore) => voters
 			.iter()
-			.find_map(|(p, _)| keystore.read().key_pair::<AuthorityPair>(&p).ok()),
+			.find_map(|(p, _)| {
+				let can_vote = keystore.read()
+					.has_keys(&[(p.to_raw_vec(), AuthorityId::ID)]);
+				if can_vote {
+					Some(p.clone())
+				} else {
+					None
+				}
+			}),
 		None => None,
 	}
 }
 
 /// Returns the authority id of this node, if available.
 fn authority_id<'a, I>(
 	authorities: &mut I,
-	keystore: &Option<KeyStorePtr>,
+	keystore: &Option<BareCryptoStorePtr>,
```

Comment:
Same as above for `is_voter`. Not a problem introduced in this PR but I guess we can clean it up while we're at it.